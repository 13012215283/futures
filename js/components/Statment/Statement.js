import React from 'react';
import { View, Text, StyleSheet, TouchableOpacity } from 'react-native';
import PropTypes from 'prop-types';
import { colors, sizes } from '../themes/index';

import { px2dp } from '../../utils/px2dp';
import noop from '../../utils/noop';

const Statement = ({ isRead, children, onPress }) => (
  <View style={styles.container}>
    <TouchableOpacity style={{ marginRight: px2dp(5) }} onPress={onPress}>
      <Text
        style={{
          fontSize: px2dp(20),
          color: isRead ? colors[1004] : colors[1102],
          fontFamily: 'iconfont',
        }}
      >
        &#xe6d6;
      </Text>
    </TouchableOpacity>
    <View>
      <Text style={styles.stateText}>
        {React.Children.map(children, child => child)}
      </Text>
    </View>
  </View>
);

export default Statement;

const styles = StyleSheet.create({
  container: {
    marginTop: px2dp(14),
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
  stateText: {
    textAlign: 'center',
    fontSize: sizes.f0,
  },
});

Statement.propTypes = {
  isRead: PropTypes.bool,
  children: PropTypes.node.isRequired,
  onPress: PropTypes.func,
};

Statement.defaultProps = {
  isRead: true,
  onPress: noop,
};
