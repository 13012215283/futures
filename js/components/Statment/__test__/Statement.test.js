import React from 'react';
import { shallow } from 'enzyme';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';
import Statement from '../Statement';

describe('<Statement/>', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Statement>
          <Text>hello</Text>
        </Statement>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders multiple children correctly', () => {
    const tree = shallow(
      <Statement>
        <Text>hello</Text>
        <Text>world</Text>
      </Statement>
    );
    expect(tree.find(Text).length).toBe(4);
  });

  it('simulates events correctly', () => {
    let click = false;
    const tree = shallow(
      <Statement
        onPress={() => {
          click = true;
        }}
      >
        <Text>hello</Text>
        <Text>world</Text>
      </Statement>
    );
    tree.childAt(0).simulate('press');
    expect(click).toBe(true);
  });
});
