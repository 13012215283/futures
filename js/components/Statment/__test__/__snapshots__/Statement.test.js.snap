// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Statement/> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginTop": 14,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginRight": 5,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#0040ff",
          "fontFamily": "iconfont",
          "fontSize": 20,
        }
      }
    >
      î›–
    </Text>
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 20,
          "textAlign": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        hello
      </Text>
    </Text>
  </View>
</View>
`;
